src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -2016,38 +2016,52 @@ void ObjectMgr::LoadCreatures()
         Field* fields = result->Fetch();
 
         ObjectGuid::LowType spawnId     = fields[0].GetUInt32();
-        uint32 entry                    = fields[1].GetUInt32();
+        uint32 id                       = fields[1].GetUInt32();
+        uint32 id2                      = fields[2].GetUInt32();
+        uint32 chance                   = fields[3].GetUInt32();
 
-        CreatureTemplate const* cInfo = GetCreatureTemplate(entry);
+        CreatureTemplate const* cInfo = GetCreatureTemplate(id);
         if (!cInfo)
         {
-            LOG_ERROR("sql.sql", "Table `creature` has creature (SpawnId: %u) with non existing creature entry %u, skipped.", spawnId, entry);
+            LOG_ERROR("sql.sql", "Table `creature` has creature (SpawnId: %u) with non existing creature entry %u in creature_id1 field, skipped.", spawnId, id);
+            continue;
+        }
+        CreatureTemplate const* cInfo2 = GetCreatureTemplate(id2);
+        if (!cInfo2 && id2)
+        {
+            LOG_ERROR("sql.sql", "Table `creature` has creature (SpawnId: %u) with non existing creature entry %u in creature_id2 field, skipped.", spawnId, id2);
+            continue;
+        }
+        if (chance <= 0 || chance > 100)
+        {
+            LOG_ERROR("sql.sql", "Table `creature` chance_id1 (Value: %u) must be greater than 0 and less than or equal to 100, skipped.", chance);
             continue;
         }
 
```

Comment:
```suggestion
        if (chance == 100 && id2)
        {
            LOG_ERROR("sql.sql", "Table `creature` has spawnid = %u chance = 100 even though creature_id2 has an entry. changed to 50.", chance);
            chance = 50;
```

Reply:
chance can never be 100 if we have a creature id2 in the database. Just wouldn't work, so better if we just throw an error and change the chance to 50.

I don't think we need to skip load though

Reply:
Agreed